keps/sig-cli/1441-kubectl-debug/README.md

```diff
@@ -0,0 +1,759 @@
+  - [Node Troubleshooting with Privileged Containers](#node-troubleshooting-with-privileged-containers)
+
+Additionally we'll review unit tests to ensure they're complete before
+graduation.
+
+#### GA milestones
+
+If the `EphemeralContainers` feature has reached beta, we will add an
+integration test for [Pod Troubleshooting with Ephemeral Debug Container
+](#pod-troubleshooting-with-ephemeral-debug-container).
+
+### Graduation Criteria
+
+#### Alpha -> Beta Graduation
+
+- [x] A task on https://kubernetes.io/docs/tasks/ describes how to troubleshoot
+  a running pod using Ephemeral Containers.
+- [ ] A survey sent to early adopters doesn't reveal any major shortcomings.
+- [x] Test plan is amended to address the most common user journeys.
+- [ ] Test plan Beta milestones reached.
+
+#### Beta -> GA Graduation
+
+- [ ] Test plan GA milestones reached
+- [ ] User feedback gathered over 2 release cycles.
+- [ ] 3 external articles suggest using `kubectl debug`
+
+### Upgrade / Downgrade Strategy
+
+This functionality is contained entirely within `kubectl` and shares its
+strategy. No configuration changes are required.
+
+### Version Skew Strategy
+
+`kubectl debug` makes use the following recent features:
+
+- Process namespace sharing (alpha: 1.10, beta: 1.12, GA: 1.17)
+- Ephemeral containers (alpha: 1.16)
+
+Not all functionality in `kubectl debug` requires these features. If an
+invocation requires a feature that is not enabled in the cluster, the api server
+will reject the pod creation request.
+
+Special consideration is given to the Ephemeral Containers feature since this
+feature will not be enabled on most clusters while it is in alpha. The `kubectl
+debug -h` displays `(requires the EphemeralContainers feature to be enabled in
+the cluster)` for examples that require ephemeral containers.
+
+Since ephemeral containers use a dedicated subresource, the api server will
+return a 404 when the feature is disabled. When this happens for a target that
+exists, kubectl prints `ephemeral containers are disabled for this cluster`.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name:
+    - Components depending on the feature gate:
+  - [x] Other
+    - Describe the mechanism: **A new command in `kubectl alpha`**
+    - Will enabling / disabling the feature require downtime of the control
+      plane? **no**
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? **no**
+
+* **Does enabling the feature change any default behavior?**
+  Any change of default behavior may be surprising to users or break existing
+  automations, so be extremely careful here.
+
+  **It's a new command with no default behavior.**
```

Comment:
Nit: remove the comment above and leave just the question. And do not format your answer.